server/plugin/command/create_event.go

```diff
@@ -26,121 +32,100 @@ type userError struct {
-	event := &remote.Event{
-		Subject: "TestSubject",
-		// BodyPreview: "testBodyPreview",
-		BodyPreview: "DEBUG_BodyPreview",
-		Body: &remote.ItemBody{
-			Content:     "Hello!  Here is the start of Your Body!",
-			ContentType: "Text",
-		},
-		ReminderMinutesBeforeStart: 15,
-		Location: &remote.Location{
-			DisplayName:  "Las Vegas",
-			LocationType: "homeAddress",
-			Address: &remote.Address{
-				Street:          "3730 Las Vegas Blvd S",
-				City:            "Las Vegas",
-				State:           "Nevada",
-				CountryOrRegion: "US",
-				PostalCode:      "89158",
-			},
-			Coordinates: &remote.Coordinates{
-				Latitude:  47.672,
-				Longitude: -102.103,
-			},
-		},
-		Attendees: attendees,
-		Start: &remote.DateTime{
-			TimeZone: "Pacific Standard Time",
-			DateTime: time.Now().Format(time.RFC3339),
-		},
-		End: &remote.DateTime{
-			TimeZone: "Pacific Standard Time",
-			DateTime: time.Now().Add(time.Hour).Format(time.RFC3339),
-		},
-	}
-
-	// parse flags and start overriding Demo Defaults
+	event := &remote.Event{}
+
 	createFlagSet := getCreateEventFlagSet()
 	err := createFlagSet.Parse(args)
 	if err != nil {
 		return event, nil, err
 	}
 
-	//                //
-	// Required Flags //
-	//                //
+	help, err := createFlagSet.GetBool("help")
+	if help == true {
+		return nil, &userError{ErrorMessage: fmt.Sprintf(getCreateEventFlagSet().FlagUsages())}, nil
+	}
+
 	subject, err := createFlagSet.GetString("subject")
 	if err != nil {
 		return event, nil, err
 	}
 	// check that next arg is not a flag "--"
-	if subject == "" || strings.HasPrefix(subject, "--") {
+	if strings.HasPrefix(subject, "--") {
 		return event, &userError{ErrorMessage: "must specify an event subject"}, nil
 	}
 	event.Subject = subject
 
-	//                //
-	// Optional Flags //
-	//                //
+	body, err := createFlagSet.GetString("body")
+	if err != nil {
+		return event, nil, err
+	}
+	// check that next arg is not a flag "--"
+	if strings.HasPrefix(body, "--") {
+		return event, &userError{ErrorMessage: "must specify an event body"}, nil
+	}
+	event.Body = &remote.ItemBody{
+		Content: body,
+	}
+
 	startTime, err := createFlagSet.GetString("starttime")
 	if err != nil {
 		return event, nil, err
 	}
 	if strings.HasPrefix(startTime, "--") {
-		return event, &userError{ErrorMessage: "must specify an event subject"}, nil
+		return event, &userError{ErrorMessage: "must specify an starttime"}, nil
+	}
+	event.Start = &remote.DateTime{
+		DateTime: startTime,
+		TimeZone: "Pacific Standard Time",
 	}
-	event.Start.DateTime = startTime
 
 	endTime, err := createFlagSet.GetString("endtime")
 	if err != nil {
 		return event, nil, err
 	}
 	if strings.HasPrefix(endTime, "--") {
-		return event, &userError{ErrorMessage: "must specify an event subject"}, nil
+		return event, &userError{ErrorMessage: "must specify an endtime"}, nil
+	}
+	event.End = &remote.DateTime{
+		DateTime: endTime,
+		TimeZone: "Pacific Standard Time",
```

Comment:
This should be extracted from t.Location information. Please file a separate ticket to handle the user time zone properly - the meeting should be created in the local timezone.

Reply:
Agreed. I filed a separate ticket.

https://mattermost.atlassian.net/browse/MM-21391

Reply:
Ticket filed here: 
https://mattermost.atlassian.net/browse/MM-21391